### PR TITLE
Completely rewrote Grades command to utilize python data libraries

### PR DESCRIPTION
**Pull request by Jay-Madden**
This has a smaller memory footprint, a faster startup time, faster runtime and provides more complete information

### CODE REVIEW

File: bot/cogs/grades_cog/grades_cog.py [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+            #we need to create the columns on the page so chunk the list again
+            content = ''
+            for col in self.chunk_list(chunk, 2):
+                #the columns wont have the perfect number of elements every time, we need to append spaces if
+                #the list entries is less then the number of columns
+                while len(col) <3:
+                    col.append(' ')
+
+                #Cocatenate the formatted column string to the page content string
```
Comment (svengeance) [inline bot/cogs/grades_cog/grades_cog.py:238]: Typo here - concatenate?

File: bot/cogs/grades_cog/grades_cog.py [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+                content += "{: <24}  {: <24}\n".format(*col)
+
+            #Apped the content string to the list of pages to send to the paginator
```
Comment (svengeance) [inline bot/cogs/grades_cog/grades_cog.py:241]: Typo here - append?

**Review by Emimendoza — CHANGES_REQUESTED** [review]

File: bot/cogs/grades_cog/grades_cog.py [hunk 3]
```diff
@@ -0,0 +1,265 @@
+import json
+import os
+import gzip
+import logging
+import collections
+import itertools
+
+import discord
+import pandas as pd
```
[the remark this answers is not in the document]
Comment (Jay-Madden) [reply bot/cogs/grades_cog/grades_cog.py:9]: its been updated mephi

[the remark this answers is not in the document]
Comment (Emimendoza) [reply bot/cogs/grades_cog/grades_cog.py:9]: Alright

File: bot/cogs/grades_cog/grades_cog.py [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+class GradesCog(commands.Cog):
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.load_data()
+
+    def load_data(self):
+        self.grades_df = pd.DataFrame()
+        for file in CSV_FILES:
+            self.grades_df = self.grades_df.append(pd.read_csv(f'{ASSET_LOCATION}{file}'))
```
Comment (zepthro) [inline bot/cogs/grades_cog/grades_cog.py:51]: So I think we have a few unnecessary additions. panda.read_csv can infer the .csv files as being zipped or not. Might be worthwhile keeping the csv's zipped. 
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html
_under compression_

Comment (Jay-Madden) [reply bot/cogs/grades_cog/grades_cog.py:51]: I did actually consider this, but due to the fact the new CSV files are only a few hundred kilobytes, you don't gain much from zipping them and keeping them human readable is valuable

File: bot/cogs/grades_cog/grades_cog.py [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+    '2019_Fall.csv',
+    '2020_Fall.csv',
+    '2014_Spring.csv',
+    '2015_Spring.csv',
+    '2015_Spring.csv',
+    '2016_Spring.csv',
+    '2017_Spring.csv',
+    '2018_Spring.csv',
+    '2019_Spring.csv',
+]
```
[the remark this answers is not in the document]
Comment (Jay-Madden) [reply bot/cogs/grades_cog/grades_cog.py:38]: This is another thing I've grappled with, we could do a os.listdir call that would bring it inline with the rest of the bot

[the remark this answers is not in the document]
Comment (Emimendoza) [reply bot/cogs/grades_cog/grades_cog.py:38]: I think that would be a better solution so to add more data for future years will be basically drag and drop without needing to modify any code

[the remark this answers is not in the document]
Comment (smathep) [reply bot/cogs/grades_cog/grades_cog.py:38]: I think thats a good solution. Less upkeep to do in the future, even though it is a bit simple.

File: bot/cogs/grades_cog/grades_cog.py [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+        Attempts to give more information about professor's grade distribution at Clemson.
+
+        DISCLAIMER:
+        Not every professor listed will be at Clemson, this is a tool built for better information but not complete information.
+        In addition, this system works on the Grade Distribution Releases located at https://www.clemson.edu/institutional-effectiveness/oir/data-reports/
+        In Addition, Clemson provides incomplete and mangled professor data so there will be multiple different versions of the same professor. This is just a byproduct of how the data is distributed from the university
+        """
+    )
+    @ext.short_help('Provides info about a given professor')
+    @ext.example(('prof brian dean', 'prof brian dean'))
```
Comment (smathep) [inline bot/cogs/grades_cog/grades_cog.py:135]: Is there a reason to have Brain Dean listed as an example twice in the help documentation?

File: scripts/normalize_grade_data.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+Not a complete reference of possible invalid data
+
+"""
+file = sys.argv[1]
+
+if '.csv' not in file:
+    exit('File must be a csv')
+
+if '2020' in file:
+    #2020 added some course designatins so we need to make extra columns
```
Comment (smathep) [inline scripts/normalize_grade_data.py:16]: typo on "designatins"

File: bot/cogs/grades_cog/grades_cog.py (hunk 6 above)
Comment (Jay-Madden) [reply bot/cogs/grades_cog/grades_cog.py:135]: no i messed up the copy pasting

**Pull request merged**